### ISSUE TITLE
UnitTests: TeleportationIdentityTestHelper is missing Reset() ?

### ISSUE THREAD

**Issue by StefH** [Microsoft/Quantum#37]
When running the unit-tests for **Teleportation**, I get this error:

> Message: Microsoft.Quantum.Simulation.Simulators.Exceptions.ReleasedQubitsAreNotInZeroState : Released qubits are not in zero state.

Solution is to add a `ResetAll(anc);` statement at line
https://github.com/Microsoft/Quantum/blob/master/Samples/UnitTesting/TeleportationTests.qs#L19

**Comment by anpaz-msft** [comment]
Hi StefH,

The test as-is today should be working ok. 
Inside the Teleporation method, we measure and reset the `source` qubit: https://github.com/Microsoft/Quantum/blob/bf27332e45a55b29db41f20b0bd9d466497ceb17/Samples/UnitTesting/Teleportation.qs#L71
so the `SWAP` should be taking it back to a `Zero` state, so `ResetAll` should not be needed. 

Are you seeing instances where this is not the case?

**Comment by StefH** [comment]
Actually I wonder if this specific test is included in the test-run ? Which C# file starts this test ?

**Comment by anpaz-msft** [comment]
Yes, it is. This one, as most of other Q# tests from this project are discovered and executed by the `SimulatorTestTargets`:
https://github.com/Microsoft/Quantum/blob/bf27332e45a55b29db41f20b0bd9d466497ceb17/Samples/UnitTesting/QuantumSimulatorTestTargets.cs#L36
